apps/linkedin-to-notion/src/contents/linkedin-notion-side-panel.tsx

```diff
@@ -24,27 +27,39 @@ export const getStyle: PlasmoGetStyle = () => {
 const LinkedinNotionSidePanel = () => {
   const [isOpen, setIsOpen] = useStorage('linkedInNotionSidePanelIsOpen', false);
 
-  const [notionToken, setNotionToken] = useStorage<{
-    refreshToken: string;
-    accessToken: string;
-  }>('notionToken');
-  notionToken; // to remove ts error
-
-  const [selectedNotionDatabase, setSelectedNotionDatabase] = useStorage<string>('selectedNotionDatabase');
+  const [selectedNotionDatabase, setSelectedNotionDatabase] = useStorage<string | null>('selectedNotionDatabase');
   selectedNotionDatabase; // to remove ts error
 
+  const [user, setUser] = useStorage<Tables<'users'>>('user');
+  const [session, setSession] = useStorage<Session | null>('session');
+
   useEffect(() => {
     const initSession = async () => {
       const { data } = await supabase.auth.getSession();
+      setSession(data.session);
+    };
 
-      setNotionToken({
-        accessToken: data.session.provider_token,
-        refreshToken: data.session.refresh_token,
+    const initUser = async () => {
+      const userData = await sendToBackground<
+        {
+          authenticatedUserId: string;
+        },
+        Tables<'users'>
+      >({
+        name: 'users/resolvers/getOrCreateUserWithAuthenticatedUserId',
+        body: { authenticatedUserId: session.user.id },
       });
+      setUser(userData);
     };
 
-    initSession();
-  }, []);
+    if (!session || session?.expires_at < Date.now()) {
```

Comment:
```suggestion
    if (!session) {
```

see comment on `onboarding.tsx`